File: src/csharp/SlimShader/Chunks/Common/PrimitiveTopology.cs
namespace SlimShader.Chunks.Common
{
	public enum PrimitiveTopology
	{
		Undefined = 0,

		[Description("pointlist", ChunkType.Shex)]
		PointList = 1,

		[Description("linelist", ChunkType.Shex)]
		LineList = 2,

		[Description("linestrip", ChunkType.Shex)]
		LineStrip = 3,

		[Description("trianglelist", ChunkType.Shex)]
		TriangleList = 4,

		[Description("trianglestrip", ChunkType.Shex)]
		TriangleStrip = 5,

		// 6 is reserved for legacy triangle fans
		// Adjacency values should be equal to (0x8 & non-adjacency):

		LineListAdj = 10,
		LineStripAdj = 11,
		TriangleListAdj = 12,
		TriangleStripAdj = 13,
	}
}

File: src/cpp/Source/Chunks/Common/PrimitiveTopology.cpp
#include "PCH.h"
#include "PrimitiveTopology.h"

#include "Decoder.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToStringShex(PrimitiveTopology value)
{
	switch (value)
	{
	case PrimitiveTopology::PointList :
		return "pointlist";
	case PrimitiveTopology::LineList :
		return "linelist";
	case PrimitiveTopology::LineStrip :
		return "linestrip";
	case PrimitiveTopology::TriangleList :
		return "trianglelist";
	case PrimitiveTopology::TriangleStrip :
		return "trianglestrip";
	default :
		throw runtime_error("Unsupported value: " + to_string((int) value));
	}
}